### PR TITLE
Rocky/provider fixes

### PR DESCRIPTION
**Pull request by rocky-fleek**
## Summary
- Added getPrincipal to interface
- Added agent reinitialization on createActor
- Added recursive bigint parsing to prevent crashes during comunications

### CODE REVIEW

File: src/Provider.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ interface CreateActor<T> {
 
 interface RequestConnectParams {
   whitelist: string[];
-  host: string;
```
Comment (maksym-boytsov) [inline src/Provider.ts:50]: I believe Ale removed that interface completely

File: src/utils/sign.ts [hunk 2]
```diff
@@ -36,6 +36,25 @@ export const canDecodeArgs = (
   );
 };
 
+export const recursiveParseBigint = obj => {
```
Comment (maksym-boytsov) [inline src/utils/sign.ts]: it'd be nice to have it under `utils/bigint.ts`

**Review by maksym-boytsov — APPROVED** [review]
Looks great overall, small tweaks proposed

File: src/Provider.ts (hunk 1 above)
Comment (tomiir) [reply src/Provider.ts:50]: he did change it but I don't think he removed it. In any case let's handle that when we get there

**Pull request merged**